### PR TITLE
Made the bash scripts compatible with Mac OS X

### PR DESCRIPTION
**Pull request by sompylasar**
* Used a custom implementation of `readlink -e` / `readlink -f` (Mac OS
X `readlink` does not support these CLI flags).
* Wrapped all paths in commands into quotes (they in general can
contain spaces). Except globbing patterns, they cannot be quoted and
may break on paths with spaces, be careful.

### CODE REVIEW

File: KnowSheet/scripts/coverage-report.sh [hunk 1]
```diff
@@ -7,23 +7,26 @@ set -u -e
 CPPFLAGS="-std=c++11 -g -Wall -W -fprofile-arcs -ftest-coverage -DBRICKS_COVERAGE_REPORT_MODE"
 LDFLAGS="-pthread"
 
-TMPDIR=.noshit
+SCRIPT_DIR=$( $( dirname "${BASH_SOURCE[0]}" )/KnowSheetReadlinkDir.sh "$PWD" )
 
-mkdir -p $TMPDIR
+TMPDIR=".noshit"
+TMPDIR_FULL_PATH="$SCRIPT_DIR/.noshit"
```
Comment (dkorolev) [inline KnowSheet/scripts/coverage-report.sh]: This directory should be relative to the directory where the script is run from, ex. `net/api`, not the `coverage-report.sh` script, which is `Bricks/KnowSheet/scripts/`.

File: KnowSheet/scripts/full-test.sh [hunk 2]
```diff
@@ -11,34 +11,37 @@ set -u -e
...
 
-TMPDIR=.noshit/fulltest
-TMPDIR_FULL_PATH=$(readlink -e .)/.noshit/fulltest
+SCRIPT_DIR=$( $( dirname "${BASH_SOURCE[0]}" )/KnowSheetReadlinkDir.sh "${BASH_SOURCE[0]}" )
+RUN_DIR="$PWD"
 
-GOLDEN_SUBDIR_NAME=golden
-GOLDEN_FULL_PATH=$TMPDIR_FULL_PATH/$GOLDEN_SUBDIR_NAME
+TMPDIR=".noshit/fulltest"
+TMPDIR_FULL_PATH="$SCRIPT_DIR/.noshit/fulltest"
```
Comment (dkorolev) [inline KnowSheet/scripts/full-test.sh]: Same here: Relative to the current directory, not the one the script is located at.

**Comment by dkorolev** [comment]
Thanks a lot for doing this @sompylasar !

Did you compile/run tests? I'd expect some code in `net/` would require fixing, but, from the top of my head, `make docu` should work from the top-level directory of the repository, and `make` and `make check` should work from most sub-directories, like `time`, `strings`, `dflags`, `rtti`, `mq`, etc.

**Comment by sompylasar** [comment]
OK, I'll fix the temp dir location as soon as I get back to this.

The tests have passed except the ones with the compilation errors outlined
in the logs (though there were no explicit test failures for them in the
logs).

I successfully did `make README.md`, too.

**Comment by deathbaba** [comment]
LGTM

**Comment by dkorolev** [comment]
CC @mzhurovich

This PR is a bit old, probably not worth merging "as is". Nonetheless, @sompylasar did a great job putting the script together which might help. See also a few of my comments, mostly on current directory vs. script directory.

**Comment by sompylasar** [comment]
@dkorolev I've updated and fixed temp-dir location. PTAL

**Comment by dkorolev** [comment]
@sompylasar : Thank you! I can only take a look once I have Mac access. Hopefully, soon.

@mzhurovich : FYI.

**Comment by sompylasar** [comment]
@dkorolev There are still plenty of errors and warnings under Mac OS X 10.9.2 (Maverics) as of now ( #88, #89, #90 and other). I hope the pending #102 will resolve some.

**Comment by dkorolev** [comment]
Yes, but also introduces a few ones. Stayind tuned. Good thing there are more things on my plate today.

**Comment by mzhurovich** [comment]
I was also fixing these scripts today :(
Btw, as far as I understand, we only need real path to the current directory - how about just using 'pwd -P' instead of additional script implementing 'readlink -e/-f'?

**Comment by sompylasar** [comment]
@mzhurovich Not only to the current directory, but to the `KnowSheet/scripts` directory from the `KnowSheet/scripts/check-all-headers.sh` script that is run in a current directory other than `KnowSheet/scripts`, and from the `Makefile`s symlinked to directories other than `KnowSheet/scripts`.

**Comment by mzhurovich** [comment]
@sompylasar oops, you're right, will use your fixes then

**Comment by sompylasar** [comment]
@mzhurovich Please could you also verify that my fixes work in other environments (ubuntu, windows), as I've only ran them under macosx? Thanks.

**Comment by mzhurovich** [comment]
@sompylasar Sure.

**Comment by dkorolev** [comment]
Closing this one in favor of https://github.com/KnowSheet/Bricks/pull/104. Thank you @sompylasar & @mzhurovich!

**Pull request closed without merging**